cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -562,7 +562,12 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
                         // This was validated during normalization.
                         predicateValue = negatedPredicate.get(operator);
                         placeholder = "?";
-                        sqlParameters.add(String.valueOf(predicateValue));
+                        if (predicateValue instanceof Boolean){
```

Comment:
It's an instance of `List` so its not something we should do, since it will run the risk of later using `ArrayList` only APIs which would make it harder to migrate to a different list implementation in the future.